website/content/en/docs/reference/configuration/unit-tests.md

```diff
@@ -0,0 +1,359 @@
+---
+title: Unit testing Vector configurations
+short: Unit tests
+weight: 5
+aliases: ["/docs/reference/tests", "/docs/reference/configuration/tests"]
+---
+
+Vector enables you to [unit test] the [transforms] in your processing [pipeline]. Unit tests can
+improve the maintainability of your Vector configurations, particularly in larger and more complex
+pipelines. Unit tests in Vector work just like unit tests in most programming languages:
+
+1. Provide a set of **inputs** to a transform (or to [multiple transforms](#multiple) chained
+  together)
+1. Provide expected **outputs** from the changes made by the transform (or multiple transforms)
+1. Receive directly actionable feedback on test failures
+
+In general, unit tests can serve as a useful guardrail when running in Vector in production settings
+where you need to ensure that your topology doesn't exhibit unexpected behavior.
+
+This doc will begin with an [example](#example) unit test configuration and proceed to a more
+reference-style [guide](#configuring).
+
+## Verifying output {#verifying}
+
+You can use [Boolean expressions][boolean] written in [Vector Remap Language][vrl] (VRL) to verify
+that your test outputs are what you would expect given your test inputs. Here's an example:
+
+```toml
+[[tests.outputs.conditions]]
+type = "vrl"
+source = '''
+is_string(.message) && is_timestamp(.timestamp) && !exists(.other)
```

Comment:
We actually added some `assert`-style functions recently that I think it'd be good to use here like:

```
assert!(is_string(.message), ".message should be a string")
assert!(is_timestamp(.timestamp), ".timestamp should be a timestamp")
assert!(!exists(.other), ".other should not exist")
```

The advantages are:

* You can provide an optional failure message
* If it fails, Vector actually shows the assertion that failed. Currently, the way you have it, it could be unclear which part of the conditional failed

In that same vein, I think it'd be useful to explicitly call out `assert!()` and `assert_eq!()` in this guide as a way to write the test assertions. I might even deprecate the "returns a boolean" style tests as described here and just focus on assert-style tests.

Reply:
Ugh. I wish I'd known that this were the suggested way to write unit tests, as it basically entails rewriting most of this doc. In general, it'd probably be best for Vector engineers to write documentation like this going forward, as I simply don't have access to this information in my current role.